### PR TITLE
Improve reboot required check

### PR DESCRIPTION
**Pull request by alexmuller**
Group machines by whether they should be rebooting automatically or not.

### CODE REVIEW

File: modules/monitoring/files/usr/lib/nagios/plugins/check_reboots_required [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
 
-if [[ $exitcode -eq 0 ]]; then
-  echo "OK: no hosts needing reboot"
+# Safety check that we haven't added other safe to reboot options
+# that we don't account for here.
+unchecked_machines = all_machines - (unsafe_machines + safe_machines)
+
+if unchecked_machines.length > 0
+  puts "Some machines have a 'safe to reboot' class which isn't in check_reboots_required"
+  exit 1
```
Comment (rjw1) [inline modules/monitoring/files/usr/lib/nagios/plugins/check_reboots_required]: i wonder if this should exit 3 for unknown.

**Comment by rjw1** [comment]
:+1: my comment shouldn't stop this being merged

**Comment by alexmuller** [comment]
No you're right, I like a good Nagios exit code nitpick. Just waiting for the tests.

**Comment by surminus** [comment]
This will be infinitely useful on 2ndline :+1:

**Pull request merged**